src/Installer/finalizer/finalizer.csproj

```diff
@@ -0,0 +1,30 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>$(SdkTargetFramework)-windows</TargetFramework>
+    <OutputType>WinExe</OutputType>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(NETCoreSdkPortableRuntimeIdentifier.StartsWith('win-'))">
+    <PublishAot>true</PublishAot>
+    <UseCurrentRuntimeIdentifier>true</UseCurrentRuntimeIdentifier>
+    <_IsPublishing>true</_IsPublishing>
+    <PublishDir>$(ArtifactsDir)bin\finalizer\$(Configuration)\$(SdkTargetFramework)-win-$(Architecture)\publish</PublishDir>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="$(SdkSrcRoot)Microsoft.Win32.Msi\Microsoft.Win32.Msi.csproj" />
```

Comment:
Are you sure it's from that ProjectReference? Looking at this with [Sizoscope](https://www.nuget.org/packages/sizoscope), it feels like most of the overhead comes from the Microsoft.Deployment.DotNet.Releases PackageReference below. It pulls in ResourceManager and that pulls in the entire reflection stack - that's hundreds of kilobytes in an app that doesn't otherwise use reflection.

You can also get some extra size savings by setting `OptimizationPreference` property to `Size`, and setting the `StackTraceSupport` property to `false`.

Reply:
The ResourceManager overhead of Microsoft.Deployment.DotNet.Releases doesn't look useful in general, looks like we don't actually localize this and all the resources are in English anyway. Getting rid of ResourceManager would save a couple hundred kB.

You'll also get a ~350 kB size saving once a .NET 10 SDK with https://github.com/dotnet/runtime/pull/110671 is picked up. That PR allows saving that much on any app with Control Flow Guard enabled (it got enabled for compliance) that doesn't use server GC (this one doesn't) and sets OptimizationPreference=Size (you also now set it).